### PR TITLE
fixed login; added getActivities, getDailyHeartRate, getDailySleep

### PR DESCRIPTION
**Pull request by squeezeday**
Fixed login issues for me :)

### CODE REVIEW

**Comment by JakePartusch** [comment]
Hey, thanks @squeezeday! Sorry about the late response, I missed the notification for this. I'll try to test your changes later today!

File: index.js [hunk 1]
```diff
@@ -29,29 +29,66 @@ module.exports = class GarminNodeApi {
...
+            console.log('Login OK. Ticket: ' + ticket)
             try {
-                await axios.post('https://connect.garmin.com/post-auth/login?ticket=' + ticket);
-            } catch(e)  {
-                await axios.get(e.response.headers.location);
-            };
+                var url = 'https://connect.garmin.com/modern/?ticket=' + ticket
+                response = await axios.post(url)
+                var modernResponse = response.data;
+                var displayNameRegex = /displayName\\\":\\\"([^\"]+)\\\",/;
```
Comment (JakePartusch) [inline index.js]: Is this display name stuff necessary here? Looks like maybe it is grabbing the display name out of the html?

File: index.js [hunk 2]
```diff
@@ -29,29 +29,66 @@ module.exports = class GarminNodeApi {
...
+            }
         } else {
             console.log('Login Failed');
         }
     }
 
     async getSteps(fromDate, untilDate) {
-        var username = await this.getUsername();
-        var response = await axios.get('https://connect.garmin.com/modern/proxy/userstats-service/wellness/daily/'+ username + '?fromDate=' + fromDate + '&untilDate=' + untilDate + '&metricId=29&metricId=38&grpParentActType=false');
+        var url = 'https://connect.garmin.com/modern/proxy/userstats-service/wellness/daily/'+ this.displayName + '?fromDate=' + fromDate + '&untilDate=' + untilDate + '&metricId=29&metricId=38&grpParentActType=false';
```
Comment (JakePartusch) [inline index.js]: Template strings using backticks `` can make this a bit cleaner 
```
`http://example.com?var=${myVariable}&var2=${myVariable2}`
```

File: index.js [hunk 3]
```diff
@@ -29,29 +29,66 @@ module.exports = class GarminNodeApi {
...
+        var response = await axios.get(url);
         return response.data;
     }
 
     async getUsername() {
-        var response = await  axios.get('https://connect.garmin.com/user/username');
-        return response.data.username;
+        try {
+            var url = 'https://connect.garmin.com/modern/'
```
Comment (JakePartusch) [inline index.js]: Looks like we can more easily get the username with `https://connect.garmin.com/modern/currentuser-service/user/info`

File: package.json [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
   "name": "garmin-node-api",
-  "version": "0.0.1",
+  "version": "0.0.2",
   "description": "Module for logging in to Garmin Connect and retrieving basic data",
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1"
   },
-  "author": "Jake Partusch",
+  "author": "Jake Partusch, Michael Svanström",
```
Comment (JakePartusch) [inline package.json:9]: I'd be happy to add you under "contributors" :)

https://docs.npmjs.com/files/package.json#people-fields-author-contributors

File: index.js [hunk 5]
```diff
@@ -29,29 +29,66 @@ module.exports = class GarminNodeApi {
...
+            var displayNameRegex = /displayName\\\":\\\"([^\"]+)\\\",/;
+            return modernResponse.match(displayNameRegex)[1];
+        } catch (e)
+        {
+            console.log(e)
+        }
+        return null;
+    }
+
+    async getActivities(fromDate, untilDate)
```
Comment (JakePartusch) [inline index.js:63]: Thanks for adding these new features!

File: index.js [hunk 6]
```diff
@@ -29,29 +29,66 @@ module.exports = class GarminNodeApi {
...
+    {
+        var url = 'https://connect.garmin.com/modern/proxy/userstats-service/activities/daily/'+this.displayName+'?fromDate='+fromDate+'&untilDate='+untilDate+'&metricId=17&grpParentActType=true';
+        var response = await axios.get(url)
+        return response.data;
+    }
+
+    async getDailyHeartRate(date)
+    {
+        var url = 'https://connect.garmin.com/modern/proxy/wellness-service/wellness/dailyHeartRate/'+this.displayName+'?date='+date;
+        var response = await axios.get(url)
```
Comment (JakePartusch) [inline index.js]: In general, prefer `const` to `var`/`let`. ( I know that isn't consistent throughout the rest of the code ) :)

**Comment by squeezeday** [comment]
Thanks @JakePartusch for great feedback. I've made a few changed according to your comments. Hope you see them fit.

How did you find out about the endpoint https://connect.garmin.com/modern/currentuser-service/user/info for getting the username?

**Pull request merged**